git_config.go

```diff
@@ -275,15 +275,22 @@ func (gc *GitConfig) GetInitTemplateDir() (templateDir string, err error) {
 }
 
 func (gc *GitConfig) SetInitTemplateDir(path string) (err error) {
-	if err = gc.setUnnamespacedKey("init.templatedir", path); err != nil {
+	if err = gc.SetUnnamespacedKey("init.templatedir", path); err != nil {
 		return err
 	}
 	return nil
 }
 
-func (gc *GitConfig) getKey(key string) (value string, err error) {
+func (gc *GitConfig) SetUnnamespacedKey(key, value string) (err error) {
```

Comment:
I think, if we decide to keep the setting of the trailer config above, I'd prefer to see that exposed as a new method that abstracts away how it is setting the value (e.g. what you have for `SetInitTemplateDir`) rather than exposing this method.

Reply:
True, that would have been better! I reverted the changes to this file to the previous commit since we decide to not set `trailer.ifexists` in the hook.